code/datums/brain_damage/imaginary_friend.dm

```diff
@@ -30,6 +30,10 @@
 	if(!friend.client && friend_initialized)
 		addtimer(CALLBACK(src, PROC_REF(reroll_friend)), 1 MINUTES)
 
+	if(HAS_TRAIT(owner, TRAIT_PERCEPTUAL_TRAUMA_BYPASS))
+		QDEL_NULL(friend)
+		RegisterSignal(owner, SIGNAL_REMOVETRAIT(TRAIT_PERCEPTUAL_TRAUMA_BYPASS), PROC_REF(reroll_friend))
```

Comment:
Seems weird to do this `on_life` - Why not register for `SIGNAL_ADDTRAIT(TRAIT_PERCEPTUAL_TRAUMA_BYPASS)` in `on_gain`? It'd be more responsive that way

Reply:
Also how does this even like, work? I see that this qdels our friend, then next life tick, if friend is null the trauma qdels itself. So is this just a complex way to cure the trauma?

Reply:
fixedd